File: backend/app/routers/activity_log.py
from fastapi import APIRouter, Depends
from sqlalchemy.orm import Session

from app.core.database import get_db
from app.crud import audit_log as audit_log_crud
from app.models.agent import SupportAgent
from app.schemas.audit_log import ActivityLogResponse
from app.services.auth_service import require_team_lead

# Team-lead only - not a general agent screen.
router = APIRouter(prefix="/activity-log", tags=["activity-log"])


@router.get("", response_model=list[ActivityLogResponse])
def list_activity(
    entity_type: str | None = None,
    entity_id: str | None = None,
    limit: int = 100,
    offset: int = 0,
    db: Session = Depends(get_db),
    _lead: SupportAgent = Depends(require_team_lead),
):
    return audit_log_crud.list_activity(
        db, entity_type=entity_type, entity_id=entity_id, limit=limit, offset=offset
    )

File: frontend/src/api/activityLog.ts
import { apiFetch } from './client';
import type { ActivityLogResponse } from '../types';

export const activityLogApi = {
  list: (params?: { entity_type?: string; entity_id?: string; limit?: number; offset?: number }) => {
    const q = new URLSearchParams();
    if (params?.entity_type) q.set('entity_type', params.entity_type);
    if (params?.entity_id) q.set('entity_id', params.entity_id);
    if (params?.limit != null) q.set('limit', String(params.limit));
    if (params?.offset != null) q.set('offset', String(params.offset));
    const qs = q.toString();
    return apiFetch<ActivityLogResponse[]>(`/activity-log${qs ? `?${qs}` : ''}`);
  },
};
